### PR TITLE
[JENKINS-32768] Round-tripping SingleSCMSource did not work

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-32768](https://issues.jenkins-ci.org/browse/JENKINS-32768)

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -44,7 +45,7 @@
   <licenses>
     <license>
       <name>The MIT license</name>
-      <url>http://www.opensource.org/licenses/mit-license.php</url>
+      <url>https://opensource.org/licenses/MIT</url>
```
Comment (jglick) [inline pom.xml]: Miscellaneous POM updates while we here.

File: pom.xml [hunk 2]
```diff
@@ -29,7 +29,8 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>2.3</version>
+    <version>2.12</version>
```
Comment (jglick) [inline pom.xml]: Cannot even do releases any more with 2.3.

File: pom.xml [hunk 3]
```diff
@@ -57,29 +58,27 @@
...
+    <developerConnection>scm:git:git@github.com:jenkinsci/${project.artifactId}-plugin.git</developerConnection>
+    <url>https://github.com/jenkinsci/${project.artifactId}-plugin</url>
     <tag>HEAD</tag>
   </scm>
 
   <properties>
-    <!-- This should be the highest common parent of the common SCM impls, e.g. Git, SVN, Hg, etc-->
-    <!-- Under this criteria, using p4-plugin's version. -->
-    <jenkins.version>1.596.1</jenkins.version>
+    <jenkins.version>1.609.3</jenkins.version>
```
Comment (jglick) [inline pom.xml]: Forced by `git` test dep.

File: src/main/resources/jenkins/scm/impl/SingleSCMSource/config-detail.jelly [hunk 4]
```diff
@@ -29,21 +29,25 @@
...
+  <!-- Known to work from CpsScmFlowDefinition/config.jelly: -->
+  <f:dropdownDescriptorSelector field="scm" title="${%Source Code Management}"/>
+  <!-- Presumably cribbed from /lib/hudson/project/config-scm.jelly but does not work for reasons TBD (and radio buttons take too much space anyway):
   <j:set var="namePrefix" value="${h.generateId()}"/>
   <f:entry title="Source Code Management">
     <table width="100%">
       <j:set var="scms" value="${descriptor.getSCMDescriptors(sourceOwner)}" />
       <j:forEach var="descriptor" items="${scms}" varStatus="loop">
-        <j:set var="scmd" value="${descriptor}" /><!-- backward compatibility with <1.238 -->
-        <f:radioBlock name="${namePrefix}.scm" value="${loop.index}" title="${scmd.displayName}" checked="${instance.getSCM().descriptor==scmd}">
```
Comment (jglick) [inline src/main/resources/jenkins/scm/impl/SingleSCMSource/config-detail.jelly]: Was referring to a nonexistent method.

**Comment by svanoort** [comment]
:bee: - looks good.

**Pull request merged**